### DISCUSSION TITLE
M2 MBA - Regularly getting painfully hot

### DISCUSSION THREAD

**Discussion by skulldogged** [tpwrules/nixos-m1#410] [Q&A]
I'm not sure if this is an Asahi issue or a NixOS-specific issue (or maybe something else) but everytime I'm using NixOS Asahi my laptop quickly gets painfully hot. btop reads out at 38°C/100°F on the CPU but just holding the laptop is painful, and the entire thing is hot, not just the area around the CPU. From what I've used of Fedora Asahi Remix it doesn't seem to be an issue on there, but I haven't used that enough to be sure. I do know that I don't have this problem on macOS. Even things as simple as idling with a few apps open (e.g. firefox, telegram, wezterm, waybar, etc.) brings the temperature up way more than it seems it should be. I haven't really seen anyone else having this issue which is even more confusing to me. Is there anything I could try to possibly mitigate this?

For reference, running with the powersave governor seems to be running at a much more acceptable temperature (29°C/84°F) but this of course means I'm stuck at 600MHz.

**Comment by tpwrules** [comment]
I do not notice that at all on my M1 Max MBP. Is graphics acceleration working properly? Please try on the latest release.

**Comment by skulldogged** [comment]
I'm currently rebuilding my system configuration based on the latest nixos-apple-silicon release so I'll keep you updated on that. I just wanted to mention something I forgot, that being I've noticed my macbook tends to "lock up" when it overheats too much (as in, the computer freezes up to the point the only fix is a force-reboot).

Regarding graphics acceleration, I'd assume it's working fine considering I have hyprland running smoothly without any issues (other than mpvpaper taking up high CPU usage?), but I'm not particularly sure how to explicitly check whether it's working.

**Comment by skulldogged** [comment]
Sorry for the late reply - I finally got around to testing with the latest release and, unfortunately, it doesn't seem to have changed anything. I forgot to mention that I checked temps under macOS and it reports much higher temperatures on everything, but my laptop isn't even warm to the touch under macOS.

**Comment by foldfree** [comment]
afaik, btop currently only provides temps for the battery sensors under linux, That is why it is lower.

I think you can check if gpu acceleration is working on mpv when doing `shift+i` while playing a video. It will show `display: gpu` or `display: llmvpipe`(aka cpu rendering).

But If you describe high cpu usage just by playing some video, most likely there is no gpu acceleration.
It's weird that even after a reinstall the issue persist.

Maybe you can try to update macOS to the latest Sonoma release and re install NixOs once again. (macOS firmware is buggy sometimes and it only shows on linux)
You could also share your `configuration.nix` so we can see if there is something wrong with it.

**Comment by malte-v** [comment]
I have a similar issue with my M2 MBA running the latest release, although the laptop only gets hot when under load (not sure if this is also the case for OP). This is particularly annoying when trying to upgrade, since lots of things need to be compiled and the macbook tends to become unresponsive even when running nixos-rebuild with `-j1`.

**Comment by Zynh0722** [comment]
I had sorta just assumed it getting toasty under load was normal :P

When playing ffxiv on macos I had similar results to when I'm recompiling everything for an upgrade.
